crates/ruff_python_parser/src/lexer.rs

```diff
@@ -508,6 +511,138 @@ impl<'source> Lexer<'source> {
+                    };
+                    self.fstring_stack.pop();
+                    return Err(LexicalError {
+                        error: LexicalErrorType::FStringError(error),
+                        location: self.offset(),
+                    });
+                }
+                '\n' if !context.allow_multiline() => {
+                    self.fstring_stack.pop();
+                    return Err(LexicalError {
+                        error: LexicalErrorType::FStringError(FStringErrorType::UnterminatedString),
+                        location: self.offset(),
+                    });
+                }
+                '\\' => {
+                    self.cursor.bump(); // '\'
+                    if matches!(self.cursor.first(), '{' | '}') {
+                        // Don't consume `{` or `}` as we want them to be consumed as tokens.
+                        break;
+                    } else if !context.is_raw_string {
+                        if self.cursor.eat_char('N') && self.cursor.eat_char('{') {
+                            in_named_unicode = true;
+                            continue;
+                        }
+                    }
+                    // Consume the escaped character.
+                    self.cursor.bump();
+                }
+                quote @ ('\'' | '"') if quote == context.quote_char() => {
+                    match context.quote_size {
+                        StringQuoteSize::Single => break,
+                        StringQuoteSize::Triple => {
+                            let mut remaining = self.cursor.rest().chars().skip(1);
+                            if remaining.next() == Some(quote) && remaining.next() == Some(quote) {
+                                break;
+                            }
+                        }
+                    }
+                    self.cursor.bump();
+                }
+                '{' => {
+                    if self.cursor.second() == '{' {
+                        self.cursor.bump();
+                        self.cursor.bump();
+                    } else {
+                        break;
+                    }
+                }
+                '}' => {
+                    if in_named_unicode {
+                        in_named_unicode = false;
+                        self.cursor.bump();
+                    } else if self.cursor.second() == '}' && !context.is_in_format_spec() {
+                        self.cursor.bump();
+                        self.cursor.bump();
+                    } else {
+                        break;
+                    }
+                }
+                _ => {
+                    self.cursor.bump();
+                }
+            }
+        }
+        let value_end = self.offset();
+
+        if value_start == value_end {
+            return Ok(None);
+        }
+
+        let value = self.source[TextRange::new(value_start, value_end)]
+            .to_string()
```

Comment:
The `to_string` call here is unnecessary. `replace` always allocates a new string already
```suggestion
```